### ISSUE TITLE
JavaScript Fundamentals

### ISSUE THREAD

**Issue by battaglr** [beerjs/cordoba#30]
Según lo que hablamos en la última BeerJS, podríamos comenzar una serie de charlas del estilo **JavaScript Fundamentals** en la que no necesariamente tratemos temas *cutting edge*, sino cosas como:

- [ ] Data Structures (New / Old)
- [x] Array Methods - @ElRodrigote
- [ ] Object Methods
- [x] Scope & Closures - @nachomena / @battaglr
- [x] `this` & Prototype - @emiacc
- [x] Async: Callbacks => Promises => Async/Await - @joelalejandro
- [x] Classes - @HGARZON
- [ ] Generators
- [ ] Modules
- [ ] ES6 Syntax Sugar / Small Features
   - [ ] Template Literals
   - [ ] Default Parameters
   - [ ] Destructuring
   - [ ] *etc.*

¿Cómo lo ven? Pueden votar con 👍 o 👎 y agregar comentarios con los que tengan ganas de tomar algún tema particular o agregar uno a la lista —yo voy editando el *issue*.

**Comment by battaglr** [comment]
Yo me ofrezco ✋ para dar una de **Array Methods**.

**Comment by joelalejandro** [comment]
Aquí voluntario para **Promises**! 👐

**Comment by martinbavio** [comment]
Me parece que Promises debería ser parte de una charla más grande
relacionada a Javascript asincronico. Me imagino algo que empiece con la
historia (callback hell) y termine con el estupendo estado actual
(async/await). Que opinan?

**Comment by joelalejandro** [comment]
@mbavio Sí, definitivamente. Imaginate una versión actualizada de esto: http://slides.com/joelalejandrovillarrealbertoldi/javascript-promises#/

**Comment by battaglr** [comment]
@mdottavio, @joelalejandro, ahí actualicé el topic a **Async: Callbacks => Promises => Async/Await**.

**Comment by ataschz** [comment]
Seria algo asi como un JavaScirpt 101????? En Baires se hace y conovoca mucha gente, pero o no lo haria en conjunto conl a BeerJS, porque esta ultima tiene la mision de ir enfocada a desarrolladores con un toque mas de senioriti, mas bien haria algo como un JavaScript 101 By BeerJs Community, y ahi mandaria cumbia, sobre todo para mantener el scope de las charlas bien definido, y no comenzar a mezclar.

**Comment by ataschz** [comment]
yo me anoto para una charla de **ES6 Syntax Sugar / Small Fetures**.

**Comment by battaglr** [comment]
@atasanchez, en JavaScript 101 se juntan una tarde y enseñan introducción a la programación usando JavaScript como lenguaje. Es decir, explican qué es una variable, qué es un loop o un array.

Lo que planteo es un poco diferente a eso. Si bien algunos temas parecen básicos, hay mucha gente que no conoce la potencia de `map` o `reduce`, que usa Promises pero no comprende cómo funcionan realmente o que no saben para qué usar un Generator (yo por ejemplo). Por eso los temas propuestos van por el lado de ir haciendo un poco de _level-up_ y mantenimiento siempre las charlas variadas que hay hasta ahora. Esto es un complemento.

Edit: quizás "Fundamentals" no fue un buen título porque entiendo que pueda dar a esta confusión. Sugerencias bienvenidas.

**Comment by HGARZON** [comment]
Aporto con este repositorio que está muy bueno.

https://github.com/trekhleb/javascript-algorithms#readme

**Comment by HGARZON** [comment]
Si puedo participar me gustaría **Classes** :)

**Comment by battaglr** [comment]
@atasanchez, después si tenés ganas fijate de qué temas puntuales de **ES6 Syntax Sugar / Small Fetures** tenés ganas de hablar así te anoto en el issue. 😬

**Comment by mdottavio** [comment]
Les parece hacer `splice` del listado y armar un nuevo issue por topic? de esa forma ya podemos ir asignando fechas a las charlas.

**Comment by battaglr** [comment]
Sí. Si te parece podenos dejar esto abierto para que la gente vea los temas no tomados, ¿te parece?

**Comment by mdottavio** [comment]
![](https://media.giphy.com/media/vMnuZGHJfFSTe/giphy.gif)

**Comment by elmiguedev** [comment]
this & Prototype, no se incluyó, en parte al menos, en la ultima charla?

**Comment by marcomaza92** [comment]
@battaglr te acordas a qué te referías cuando pusiste `Data Structures (New / Old)`?, estamos tratando de ver a que puede pertenecer pero no damos con la tecla :P xD.
